heron/uploaders/src/java/com/twitter/heron/uploader/localfs/LocalFileSystemUploader.java

```diff
@@ -106,8 +107,8 @@ public URI uploadPackage() {
       CopyOption[] options = new CopyOption[]{StandardCopyOption.REPLACE_EXISTING};
       Files.copy(source, filePath, options);
     } catch (IOException ex) {
-      LOG.info("Unable to copy: " + source.toString() + " " + ex);
-      return null;
+      throw new UploaderException(
```

Comment:
Don't cast the caught exception down to a string, wrap it with UploaderException. Also include the dest location:
```
throw new UploaderException(
  String.format("Unable to copy file %s to %s", source.toString(), filePath),
  ex);
```